### PR TITLE
Fix use external WMS

### PR DESCRIPTION
**Pull request by rldhont**
For XYZ or WMTS layers from QGIS, the use external access was not well used.

Funded by Chateauroux Agglomération

### CODE REVIEW

File: tests/qgis-projects/tests/rasters.qgs.cfg [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            "externalWmsToggle": "True",
+            "externalAccess": {
+                "contextualWMSLegend": "0",
+                "crs": "EPSG:3857",
+                "dpiMode": "7",
+                "featureCount": "10",
+                "format": "image/png;%20mode%3D8bit",
+                "layers": "commune",
+                "styles": "d%C3%A9faut",
+                "url": "https://demo.lizmap.com/lizmap/index.php/lizmap/service?repository=miscellaneous&project=flatgeobuf&VERSION=1.3.0"
```
Comment (Gustry) [inline tests/qgis-projects/tests/rasters.qgs.cfg:193]: Note for myself, this URL demo.lizmap.com shouldn't be used in tests.

https://liz.lizmap.com/tests/ instead

**Review by Gustry — APPROVED** [review]
Only reading quickly the source code, not testing

**Comment by Gustry** [comment]
I think the issue could have been identified with #4416 on the project `user_defined_based_layer`  ?

**Comment by MundoGIS** [comment]
Boost your Lizmap performance with Qtiles

If you are struggling with slow rendering times in Lizmap due to heavy vector layers or complex styling, you should definitely check out Qtiles.

Qtiles allows you to generate WMTS  (directory structure or MBTiles) directly from QGIS. By using these pre-rendered tiles in your Lizmap project instead of dynamic WMS layers, you can significantly reduce server load and improve the user experience.

Key benefits for Lizmap users:

🚀 Unmatched Performance: We have tested these tiles with Leaflet, OpenLayers, Origo-map, and QGIS. The loading speed significantly outperforms middleware solutions like GeoWebCache, GeoServer, and MapProxy.

🎨 Custom Basemaps: Bake your complex QGIS styling into a fast, static base layer.

✅ Proven Stability: Since the tiles are pre-rendered, you remove the processing overhead from your server entirely.

Give it a try! https://github.com/nextgis/Qtiles

Let me know if you need any assistance setting this up.

BR/ Abel

**Pull request merged**